remote-api/src/test/java/org/alfresco/rest/api/impl/RulesImplTest.java

```diff
@@ -394,10 +395,136 @@ public void testSaveRules_createMultipleRules()
         assertThat(actual).isEqualTo(expected);
     }
 
+    @Test
+    public void testDeleteRuleById() {
+        given(nodesMock.validateNode(eq(RULE_SET_ID))).willReturn(ruleSetNodeRef);
+        given(nodesMock.validateNode(eq(RULE_ID))).willReturn(ruleNodeRef);
+        given(ruleServiceMock.isRuleAssociatedWithRuleSet(any(), any())).willReturn(true);
+        given(ruleServiceMock.isRuleSetAssociatedWithFolder(any(), any())).willReturn(true);
+        org.alfresco.service.cmr.rule.Rule rule = createRule(RULE_ID);
+        given(ruleServiceMock.getRule(any())).willReturn(rule);
+
+        //when
+        rules.deleteRuleById(FOLDER_NODE_ID, RULE_SET_ID, RULE_ID);
+
+        then(nodesMock).should().validateOrLookupNode(eq(FOLDER_NODE_ID), isNull());
+        then(nodesMock).should().validateNode(eq(RULE_SET_ID));
+        then(nodesMock).should().validateNode(eq(RULE_ID));
+        then(nodesMock).should().nodeMatches(eq(folderNodeRef), any(), isNull());
+        then(nodesMock).should().nodeMatches(eq(ruleSetNodeRef), any(), isNull());
+        then(nodesMock).should().nodeMatches(eq(ruleNodeRef), any(), isNull());
+        then(nodesMock).shouldHaveNoMoreInteractions();
+        then(permissionServiceMock).should().hasReadPermission(eq(folderNodeRef));
```

Comment:
I assume you need more than just folder "Read" permission to delete a rule ? 

Maybe  folder"Write" permission or something more specific ? As an aside, I also see references to "ChangePermssion" later in the PR.

Reply:
This must be an issue with the POST API too. I've only tested a user who has access so far.

Reply:
I tested DELETE API when authenticated as a user with read-only permissions and it is not allowed to remove the rule in that case. `RuleServiceImpl` remove rule (existing) code is checking the permissions (`permissionService.hasPermission(nodeRef, PermissionService.CHANGE_PERMISSIONS) == AccessStatus.ALLOWED`) as Jan noticed.
It might be better to check that in `RulesImpl` at the very beginning (considering efficiency of double permission check when permissions are okay). WDYT?

Reply:
I think it sounds like a good candidate for a separate PR (and maybe a separate ticket).